cpp/src/lance/encodings/binary.h

```diff
@@ -95,6 +103,72 @@ ::arrow::Result<std::shared_ptr<::arrow::Scalar>> VarBinaryDecoder<T>::GetScalar
   return std::make_shared<typename ::arrow::TypeTraits<T>::ScalarType>(buf);
 }
 
+template <ArrowType T>
+::arrow::Result<std::shared_ptr<::arrow::Int64Array>> VarBinaryDecoder<T>::ReadPositions(
+    int32_t start, int32_t length) const {
+  auto buf = infile_->ReadAt(position_ + start * sizeof(int64_t), (length + 1) * sizeof(int64_t));
```

Comment:
why do we need to pass in length + 1 here? Isn't length * sizeof(int64_t) the number of bytes that need to be read?

Reply:
So for an array like `["aa", "bbb", "cccc"]` , the offset array should look like `[0, 2, 5, 9]`, where `[5, 9]` is used to calculate the start and end of "cccc" string.